docs/datascience/jupyter-notebooks.md

```diff
@@ -257,26 +257,23 @@ To learn more about Git integration within VS Code, visit [Version Control in VS
 
 ## Debug a Jupyter Notebook
 
-If you need additional debug support in order to diagnose an issue in your code cells, you can export it as a Python file. Once exported as a Python file, the VS Code debugger lets you step through your code, set breakpoints, examine state, and analyze problems. Using the debugger is a helpful way to find and correct issues in notebook code. To debug your Python file:
+We support two different ways to debug a Jupyter notebook: a simpler mode that we call "Run by Line", and full debugging mode.
 
-1. In VS Code, if you haven't already, activate a Python environment in which Jupyter is installed.
+> **Note:** Both of these features require ipykernel 6+
 
-1. From your Jupyter Notebook (`.ipynb`), select the **Export** button in the main toolbar.
+### Run by Line
 
-   ![Convert Jupyter Notebook to Python file](images/jupyter/native-toolbar-export.png)
+Run by Line lets you execute a cell one line at a time, without being distracted by other VS Code debug features. To start, click the Run by Line button in the cell toolbar:
 
-   Once exported, you'll have a `.py` file with your code that you can use for debugging.
+Then, click the same button to advance by one statement. You can click the cell stop button to stop early, or the continue button in the toolbar to continue running to the end of the cell.
 
-1. After saving the `.py` file, to start the debugger, use one of the following options:
+### Debug Cell
 
-    - For the whole Notebook, open the Command Palette (`kb(workbench.action.showCommands)`) and run the **Python: Debug Current File in Python Interactive Window** command.
-    - For an individual cell, use the **Debug Cell** action that appears above the cell. The debugger specifically starts on the code in that cell. By default, **Debug Cell** steps into user code. If you want to step into non-user code, you need to uncheck **Data Science: Debug Just My Code** in the Python extension settings (`kb(workbench.action.openSettings)`).
+If you want to use the full set of debugging features supported in VS Code, such as breakpoints and the ability to step in to other cells and modules, you can use the full VS Code debugger.
 
-1. To familiarize yourself with the general debugging features of VS Code, such as inspecting variables, setting breakpoints, and other activities, review [VS Code debugging](/docs/editor/debugging.md).
-
-1. As you find issues, stop the debugger, correct your code, save the file, and start the debugger again.
-
-1. When you're satisfied that all your code is correct, use the Python Interactive window to export the Python file as a Jupyter Notebook (`.ipynb`).
+1. Start by setting any breakpoints that you need to set by clicking in the left margin of a notebook cell.
+2. Then click the "Debug Cell" button in the menu next to the run button. This will run the cell in a debug session, and will pause on your breakpoints in any code that runs, even if it is in a different cell or a .py file.
```

Comment:
I think this may need to be rephrased or we could add a screenshot of where the debug button is. Based on "click the 'debug cell' button in the menu next to the run button" a user may think this is the toolbar and get confused when they don't see a debug option

Reply:
Yeah, I'd like to have a screenshot of the Debug Cell button there :)